sdk/tables/azure-data-tables/samples/async_samples/sample_encode_dataclass_model_async.py

```diff
@@ -42,50 +51,25 @@ class Car:
     product_id: Optional[UUID] = None
     inventory_count: Optional[int] = None
     barcode: Optional[bytes] = None
-    color: Optional[str] = None
-    maker: Optional[str] = None
+    color: Optional[Color] = None
+    make: Optional[str] = None
     model: Optional[str] = None
     production_date: Optional[datetime] = None
-    big_number: Optional[int] = None
     is_second_hand: Optional[bool] = None
 
 
-class Color(str, Enum):
-    WHITE = "white"
-    GRAY = "gray"
-    BLACK = "black"
-
-
-def encode_int(value):
-    return EdmType.STRING, str(value)
-
-
-def encode_uuid(value):
-    return None, str(value)
-
-
-encoder_map = {
-    int: encode_int,
-    UUID: encode_uuid,
+# Here we will define how certain types should be encoded,
+# in this case, we want to define all Python integers to be treated
+# as int64, and we also want to provide custom encoding for the enum type 'Color'.
+encoder_map: Dict[Type, Union[EdmType, Callable[[Any], Tuple[EdmType, str]]]] = {
+    Color: lambda v: (EdmType.STRING, v.value),
+    int: EdmType.INT64,
 }
 
-
-def decode_string(value):
-    try:
-        return int(value)
-    except ValueError:
-        try:
-            return Color(value)
-        except ValueError:
-            try:
-                return UUID(value)
-            except ValueError:
-                return value
-
-
-decoder_map = {
-    EdmType.STRING: decode_string,
-}
+# Here we will define how certain types should be decoded,
+# in this case we want to define all int64 properties to be decoded
+# as Python integers, as well as custom decoding for instantiating the 'Color' type.
+decoder_map: Dict[Union[Type, EdmType], Callable[[Any], Any]] = {Color: Color, EdmType.INT64: int}
```

Comment:
Same here... the type annotation looks scarier than anything else on this line....